src/app/services/login/login.service.ts

```diff
@@ -18,10 +20,25 @@ export class LoginService {
   ) {}
 
   login(loginParams: any): Observable<any> {
+    let options: any;
+    // CSRF
+    if (environment.security === 'csrf') {
+      options = {
+        withCredentials: true,
+        responseType: 'text'
+      };
+    }
+    // JWT
+    if (environment.security === 'jwt') {
+      options = { observe: 'response' };
+    }
     return this.http.post(
       this.bo.login(), // url
-      { username: loginParams.username, password: loginParams.password }, // body
+      { j_username: loginParams.username, j_password: loginParams.password }, // body
       { responseType: 'text', observe: 'response' },
```

Comment:
You are assigning the `options` variable based on `csrf`/`jwt` condition but not passing it as a payload here.

Reply:
I'dont' understand this comment, could you please provide me more details?